### PR TITLE
[petanque] Initial commit, as an OCaml library

### PR DESCRIPTION
**Pull request by ejgallego**
This is the first version of `petanque`, suitable for merging.

TODO before merging:

- [x] basic test suite (API)
- [x] complete premises API
- [x] update README.md

### CODE REVIEW

**Comment by Alizter** [comment]
What exactly is it? I read the README but I didn't understand anything.

File: README.md [hunk 1]
```diff
@@ -182,6 +182,12 @@ fully-fledged LSP client.
 add your pre/post processing passes, for example to analyze or serialize parts
 of Coq files.
 
+### 🪄 Advanced APIs for Coq Interaction
+
+Thanks to Flèche, we provide some APIs on top of it that allow advanced uses
```
Comment (Alizter) [inline README.md:187]:
```suggestion
Thanks to Flèche, we provide some APIs on top of it that allow advanced use
```

Comment (ejgallego) [reply README.md]: Thanks!

**Comment by ejgallego** [comment]
> What exactly is it? I read the README but I didn't understand anything.

This is a start of a RL agent for Coq, on top of Flèche, indeed, we need to improve the docs a lot, but for now this is just the first step.

Thanks for checking the README, should be better now and contain a link to Gym in python

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -3,17 +3,23 @@ Contributing to Coq LSP
 
 Thank you very much for willing to contribute to coq-lsp!
 
-`coq-lsp` has two components:
+The `coq-lsp` repository contains several tightly coupled components
+in a single repository, also known as monorepos, in particular:
```
Comment (Alizter) [inline CONTRIBUTING.md:7]:
```suggestion
in a single repository, also known as a monorepo, in particular:
```

Comment (ejgallego) [reply CONTRIBUTING.md:7]: Thanks!

**Pull request merged**